pkg/client/perseshttp/request.go

```diff
@@ -319,6 +321,24 @@ type errorResponse struct {
 	Message string `json:"message"`
 }
 
+// parseErrorFromBody is trying to parse the two possible error responses that the backend can return.
+func (r *Response) parseErrorFromBody() (string, error) {
+	var err error
+	response1 := api.OAuthError{}
+	err = json.Unmarshal(r.body, &response1)
```

Comment:
here if the error message is well formatted json, then it won't return an error even if the json doesn't match the struct `OauthError`

Reply:
Ouch ok then it's a big deal.. I will verify and maybe think again of another solution. 
An error kind ? Arh

Reply:
The best would probably to use a common struct for all the errors. Oauth or Perses. 
The oauth error codes could be extended with some Perses error codes. I need to think

Reply:
Okay I find three solutions: 

### Solution1: 
Set ``message`` or ``error`` fields as "required" respectively in Perses or Oauth error response. 
To do so I would have to override the UnmarshalJSON of each and if the field is not there, return an error 

### Solution2:
Introduce error kind and add a kind field on each struct

### Solution3:
Have a single Error response object satisfying the OAuth requirements : 
- with ``error`` containing all oauth errors + a default perses one (or use the "server_error" of oauth)
- with ``error_description`` with possibly more details (optional and would correspond to the ``message`` field of the current struct


WDYT ? 
I personally prefer the 3rd one but I wanted your opinion on that

Reply:
what about having a single Error response like : 

```
message: the error message or the error_description in case we are using Oauth (mandatory field)
error: additional error (optional field).
```

I am either wondering if we should really need to put the oauth error (code?) in the error field.
Does the CLI / the client needs to know the exact error ?

Reply:
Cool, then I implemented it, I let you review that.